src/model/storage.py

```diff
@@ -15,125 +15,96 @@
+            self._load()
+        return self._model
 
+    def _load(self):
+        self._model = read_sbml_model(f"data/models/{self.model_id}.sbml.gz")
+        self._model.solver = 'cplex'
+        self._model.notes['namespace'] = self.namespace
 
-class Models(object):
-    @classmethod
-    @lru_cache()
-    def get(cls, model_id):
-        try:
-            return read_model(model_id)
-        except FileNotFoundError:
-            return None
 
-    @classmethod
-    @lru_cache()
-    def get_dict(cls, model_id):
-        model = cls.get(model_id)
-        if model is not None:
-            return model_to_dict(model)
-        return None
+MODELS = [
+    ModelMeta('iJO1366', 'ECOLX', 'bigg', 'BIOMASS_Ec_iJO1366_core_53p95M'),
+    ModelMeta('iMM904', 'YEAST', 'bigg', 'BIOMASS_SC5_notrace'),
+    ModelMeta('iMM1415', 'CRIGR', 'bigg', 'BIOMASS_mm_1_no_glygln'),
+    ModelMeta('iNJ661', 'CORGT', 'bigg', 'BIOMASS_Mtb_9_60atp'),
+    ModelMeta('iJN746', 'PSEPU', 'bigg', 'BIOMASS_KT_TEMP'),
+    ModelMeta('e_coli_core', 'ECOLX', 'bigg', 'BIOMASS_Ecoli_core_w_GAM'),
+    ModelMeta('ecYeast7', 'YEAST', 'yeast7', 'r_2111'),
+    ModelMeta('ecYeast7_proteomics', 'YEAST', 'yeast7', 'r_2111'),
+]
 
 
-def preload_cache():
-    logger.info("Preloading models")
-    with log_time(operation="Preload models"):
-        for model_id in constants.MODELS:
-            Models.get_dict(model_id)
+def get(model_id):
+    """Return the meta instance for the given model id"""
+    for model_meta in MODELS:
+        if model_meta.model_id == model_id:
+            return model_meta
+    else:
+        raise KeyError(f"No model with id '{model_id}'")
 
 
-if app.config['ENVIRONMENT'] in ['production', 'staging']:
-    preload_cache()
+def save_changes(model, message):
+    """Save adapted changes in cache based on the given model and message"""
+    changes_key = _changes_key(model.id, message)
+    value = json.dumps({'model': model.id, 'changes': model.notes.get('changes', constants.get_empty_changes())})
+    redis.set(changes_key, value)
+    logger.info(f"Stored changes for {model.id} as '{changes_key}'")
+    return changes_key
 
 
-def find_changes_in_db(model_id):
-    dumped_changes = redis.get(model_id)
-    if dumped_changes is not None:
-        return dumped_changes.decode('utf-8')
-    return None
+def restore_from_message(model_id, message):
+    """Restore model with modifications from cache based on the given model and message"""
+    return restore_from_key(_changes_key(model_id, message))
 
 
-def restore_from_db(model_id):
-    with log_time(operation=f"Restored model {model_id} from db"):
-        changes = find_changes_in_db(model_id)
-        if not changes:
-            return None
-        return model_from_changes(changes)
+def restore_from_key(changes_key):
+    """Restore model with modifications from cache based on the given unique cache key"""
+    from model.operations import restore_changes  # FIXME: circular dependency issue
```

Comment:
What if you put the import statement at the bottom of the module?

Reply:
This is actually resolved on the head of `test-optimization` and the fix will be merged as part of that branch.